chrisj117/CS-472-GROUP-2-PROJECT
# Extended Review Model

**_WHAT DOES THIS PR DO?_**

  - [x] Add a `year_taken` field to store the year the course was taken (Defaulted to Current Year).
  - [x] Add a `textbook_required` boolean field to indicate if a textbook was necessary for a course 
  - [x] Add a `recommended` boolean field to categorize reviews as recommended or not

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

  - [X] #80 

**_HOW DO I TEST OUT THIS PR?_**
Run this project locally and test the models.
First, Create a superuser for the Django admin panel if you haven't done so already:
- Run `python manage.py createsuperuser` and follow the prompts to set up your username, email, and password.
Start the Django development server by running `python manage.py runserver`.
To test the model go to the Django admin panel at  `http://localhost:8000/admin/`.
- Log in using the `superuser account` you created.
- Navigate to the Reviews section and click on Add Review to enter a new review, or select an existing one to edit.
- In the review form, you will see fields for `Year Taken`, `Textbook Required`, and `Recommended`. Fill in these details as appropriate.
- Save the review and ensure it is listed correctly with the new fields populated.
 
**Note**: The API endpoints for interacting with these new fields are currently in development. Once completed, they will provide a better way to create and modify these fields through our front end.

**_ADDITIONAL INFORMATION_**
- [x] Pipeline passing
- [x] Model Test
- [x] Code coverage 100%
- [x] Running locally
